### PR TITLE
Fix debug assertion failure in gecko CI

### PR DESCRIPTION
**Pull request by SimonSapin**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15872)

### CODE REVIEW

**Review by emilio — APPROVED** [review]

File: components/style/properties/declaration_block.rs [hunk 1]
```diff
@@ -289,9 +289,13 @@ impl PropertyDeclarationBlock {
         let important_count = &mut self.important_count;
         let mut removed_at_least_one = false;
         self.declarations.retain(|&(ref declaration, importance)| {
-            let remove = declaration.id().is_or_is_longhand_of(property);
+            let id = declaration.id();
+            let remove = id.is_or_is_longhand_of(property);
             if remove {
                 removed_at_least_one = true;
+                if let PropertyId::Longhand(id) = *property {
```
Comment (emilio) [inline components/style/properties/declaration_block.rs]: As discussed on IRC, this is wrong and needs to look at `id`, not at `property`.

r=me with this fixed.

**Review by emilio — COMMENTED** [review]
Forgot to save the comment, heh.

**Comment by emilio** [comment]
`manifest_changed` is what is preventing this from landing.

**Comment by Manishearth** [comment]
```diff
diff --git a/tests/wpt/mozilla/meta/MANIFEST.json b/tests/wpt/mozilla/meta/MANIFEST.json
index a20dffa..d554b57 100644
--- a/tests/wpt/mozilla/meta/MANIFEST.json
+++ b/tests/wpt/mozilla/meta/MANIFEST.json
@@ -20620,7 +20620,7 @@
    "support"
   ],
   "css/bug_1345483.html": [
-   "d5e1649c88102f27da5fe1ac16715cfee7f70f84",
+   "41b55af9d5c95910b1af74c0fbffb24e20bbe869",
    "testharness"
   ],
   "css/button_css_width.html": [
```

**Pull request merged**